rust/lance-index/src/vector/hnsw/builder.rs

```diff
@@ -790,7 +791,7 @@ mod tests {
         .unwrap();
 
         let object_store = ObjectStore::memory();
-        let path = object_store.base_path().child("test_builder_write_load");
+        let path = Path::from("test_builder_write_load");
```

Comment:
why this change?

Reply:
`ObjectStore::base_path` carries the same `base path` as `Dataset::base`. Logically, "base path" is not a attribute to `ObjectStore` which should be an handler for IO?